### ISSUE TITLE
support metadata in add_vertex! 

### ISSUE THREAD

**Issue by leclere** [sbromberger/MetaGraphs.jl#10]
When adding a new vertex to the MetaGraph the best way I found to give properties was the following
 ```
add_vertex!(g)

  set_props!(g, nv(g), Dict(:prop1=>prop1, :prop2 => prop2))
```

Is there a way to do this in a single line ? If not I suggest to add methods to add_vertex! to allow it.

**Comment by leclere** [comment]
If my proposed solution is working and there are no better solution I'm willing to make a PR on this issue.

**Comment by sbromberger** [comment]
This seems useful. PR offer accepted. It would be good to have a similar capability for `add_edge!` as well.

**Comment by leclere** [comment]
Hello,

I made a local version. I have two difficulties : 
- I cannot push a branch, I don't exactly know the best way to procede 
- I am a bit lost in the test file, thus my test coverage is far from perfect

**Comment by sbromberger** [comment]
Hi @leclere - the best thing to do is to fork the repository (the upper right hand corner of this page), then clone that fork to your local machine using git. Make your changes and push them to a branch in your forked copy, and then you can create a PR here pointing to that branch.

Don't worry about testing; we'll worry about that once we can see the code.

**Comment by sbromberger** [comment]
Closed via #11 - Thanks, @leclere !
